qrexec/tools/qrexec_policy_daemon.py

```diff
@@ -99,6 +99,78 @@ async def handle_client_connection(log, policy_cache,
     finally:
         writer.close()
 
+async def handle_qrexec_connection(log, policy_cache,
+                                   reader, writer):
+
+    args = {}
+
+    try:
+        untrusted_data = await reader.read(65536)
+        if len(untrusted_data) > 65535:
+            log.error('Request length too long: %d', len(data))
+            return
+        try:
+            # Qrexec guarantees that this will be present
+            service_descriptor_len = untrusted_data.index(b'\0')
```

Comment:
`service_descriptor, untrusted_data = untrusted_data.split(b'\0', 1)` ?

Reply:
Good idea.  If we really cared about performance, a faster approach would be to split at `b' '` first, then at `b'\0'`.  But the complexity probably is not worth it.